experimental/PropertyDDS/packages/property-shared-tree-interop/src/schemaConverter.ts

```
@@ -0,0 +1,362 @@
+}
+
+export function convertPSetSchemaToSharedTreeLls(
+	repository: StoredSchemaRepository,
+	rootFieldSchema: FieldSchema,
+): void {
+	const inheritedChildrenByType = loadInheritedChildren();
+	const globalTreeSchema: Map<TreeSchemaIdentifier, NamedTreeSchema> = new Map();
+	// Extract all referenced typeids for the schema
+	const unprocessedTypeIds: TreeSchemaIdentifier[] = [];
+	const rootBaseTypes = rootFieldSchema.types ?? fail("Expected root types");
+	const rootTypes = getInheritedChildrenForTypes(inheritedChildrenByType, rootBaseTypes);
+	for (const type of rootTypes) {
+		unprocessedTypeIds.push(type);
+	}
+	const unprocessedTypeIdsSet: Set<string> = new Set(unprocessedTypeIds);
+	const referencedTypeIDs = new Set<TreeSchemaIdentifier>();
+
+	while (unprocessedTypeIds.length > 0) {
+		const unprocessedTypeID = unprocessedTypeIds.pop();
+
+		if (!unprocessedTypeID) {
+			return;
+		}
+
+		referencedTypeIDs.add(unprocessedTypeID);
+
+		const schemaTemplate = PropertyFactory.getTemplate(unprocessedTypeID);
+		if (schemaTemplate === undefined) {
+			throw new Error(`Unknown typeid: ${unprocessedTypeID}`);
+		}
+		const dependencies = PropertyTemplate.extractDependencies(
+			schemaTemplate,
+		) as TreeSchemaIdentifier[];
+		for (const dependencyTypeId of dependencies) {
+			const idsSet = getInheritedChildrenForType(inheritedChildrenByType, dependencyTypeId);
+			idsSet.forEach((id) => {
+				if (!referencedTypeIDs.has(dependencyTypeId) && !unprocessedTypeIdsSet.has(id)) {
+					unprocessedTypeIds.push(id);
+					unprocessedTypeIdsSet.add(id);
+				}
+			});
+		}
+
+		// Extract context information (i.e. array, map and set types)
+		const extractContexts = (properties: any[]): void => {
+			if (properties !== undefined) {
+				for (const property of properties || []) {
+					if (property.properties) {
+						// We have a nested set of properties
+						// TODO: We have to create a corresponding nested type
+						extractContexts(property.properties);
```

Should we throw an exception "Nested types not yet supported" here? Otherwise, it will just contine, but create an invalid schema.

---

Fixed in  ac548cbbd4c5dd5f1d4d7e472d44df57a0681301